constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
+			if err := rr.AddBaseFromFs(tc.path); !errors.Is(err, tc.wantErr) {
+				t.Errorf("AddBaseFromFs() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestRegoRewriterEntryPoint(t *testing.T) {
+	tests := []struct {
+		name       string
+		prefix     string
+		content    string
+		wantResult map[string]string
+		wantError  error
+	}{
+		{
+			name:   "entry point imports lib",
+			prefix: "prefix.path",
+			content: `package lib.alpha
+				import data.lib.alpha
+        violation[{"msg":msg}] {
+          x := data.lib.alpha
+					y := data.lib.alpha.ext[_]
+          z := "data.lib.alpha"
+					data.lib.alpha.check[input.name]
+					ex := data.ext.alpha
+        }`,
+			wantResult: map[string]string{"path": `package lib.alpha
+
+import data.prefix.path.lib.alpha
+
+violation[{"msg": msg}] {
+	x := data.prefix.path.lib.alpha
+	y := data.prefix.path.lib.alpha.ext[_]
+	z := "data.lib.alpha"
+	data.prefix.path.lib.alpha.check[input.name]
+	ex := data.ext.alpha
+}
+`},
+		}, {
+			name:   "entry point binds data.lib to var",
+			prefix: "prefix",
+			content: `package templates.stuff.MyTemplateV1
+				import data.lib.alpha
+        violation[{"msg":msg}] {
+          x := data.lib
+					y := x[_]
+        }`,
+			wantResult: map[string]string{"path": `package templates.stuff.MyTemplateV1
+
+import data.prefix.lib.alpha
+
+violation[{"msg": msg}] {
+	x := data.prefix.lib
+	y := x[_]
+}
+`},
+		}, {
+			name:   "entry point uses data.lib[_]",
+			prefix: "prefix",
+			content: `package templates.stuff.MyTemplateV1
+				import data.lib.alpha
+        violation[{"msg":msg}] {
+          x := data.lib[_]
+        }`,
+			wantResult: map[string]string{"path": `package templates.stuff.MyTemplateV1
+
+import data.prefix.lib.alpha
+
+violation[{"msg": msg}] {
+	x := data.prefix.lib[_]
+}
+`},
+		}, {
+			name:   "entry point references input",
+			prefix: "prefix",
+			content: `package templates.stuff.MyTemplateV1
+         violation[{"msg":msg}] {
+          bucket := input.asset.bucket
+        }`,
+			wantResult: map[string]string{"path": `package templates.stuff.MyTemplateV1
+
+violation[{"msg": msg}] {
+	bucket := input.asset.bucket
+}
+`},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := New(NewPackagePrefixer(tc.prefix), []string{"data.lib"}, []string{"data.ext"})
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			if err := rr.AddEntryPoint("path", tc.content); err != nil {
+				if !errors.Is(err, tc.wantError) {
+					t.Fatalf("failed to add base source %q", err)
+				}
+				return
+			}
+			sources, err := rr.Rewrite()
+			if !errors.Is(err, tc.wantError) {
+				t.Errorf("Rewrite() got error = %v, want %v", err, tc.wantError)
+			}
+
+			result, err := sources.AsMap()
+			if err != nil {
+				t.Fatalf("unexpected error during Sources.AsMap: %s", err)
+			}
+			// TODO remove log
+			t.Logf("%v", result)
+
+			if diff := cmp.Diff(result, tc.wantResult); diff != "" {
```

Comment:
Standardize `cmp.Diff` to use want before got. So `cmp.Diff(tc.wantResult, result)`

Reply:
Swapped for want and got.